src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/DetachDatabase.cs

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts
+{
+    public class DetachDatabaseRequestParams : GeneralRequestDetails
+    {
+        /// <summary>
+        /// SFC (SMO) URN identifying the object  
+        /// </summary>
+        public string? ObjectUrn { get; set; }
```

Comment:
Why is this optional? When will we ever not have this?

(same for the connection URI)

Reply:
I removed "nullable disable" from the top of the file, and then compiler wanted these strings to be optional.

Reply:
You can leave nullable disabled for these files - we haven't found a good way to handle that for "contract" classes like this that isn't just clutter. 

But these should match the typings on the ADS side - where I assume you won't have these properties be optional.